src/Bicep.LangServer/Snippets/SnippetsProvider.cs

```diff
@@ -410,5 +446,56 @@ private IEnumerable<Snippet> GetRequiredPropertiesForObjectType(TypeSymbol typeS
                 }
             }
         }
+
+        public IEnumerable<Snippet> GetNestedResourceDeclarationSnippets(TypeSymbol typeSymbol)
+        {
+            // Leaving out the API version on this, because we expect its more common to inherit from the containing resource.
+            yield return new Snippet(@"resource ${1:Identifier} '${2:Type}' = {
+  name: $3
+  properties: {
+    $0
+  }
+}", prefix: "resource-with-defaults", detail: "Nested resource with defaults");
+
+            yield return new Snippet(@"resource ${1:Identifier} '${2:Type}' = {
+  name: $3
+  $0
+}", prefix: "resource-without-defaults", detail: "Nested resource without defaults");
+
+            string parentType = typeSymbol.Name;
+
+            if (resourceTypeToChildTypeSymbolsMap.ContainsKey(parentType))
```

Comment:
You can replace the `Contains()` check & dictionary access below with just a single operation:

```csharp
if (resourceTypeToChildTypeSymbolMap.TryGetValue(parentType, out var nestedTypeSymbols))
{
    foreach (var nestedTypeSymbol in nestedTypeSymbols)
    ...
```

Reply:
Updated here:
https://github.com/Azure/bicep/commit/0c0c664878dbd906644c181a547c42f854f4d5f3